asserts/snapasserts/validation_sets.go

```diff
@@ -281,3 +288,36 @@ func (v *ValidationSets) Conflict() error {
 	}
 	return nil
 }
+
+// CheckInstalledSnaps checks installed snaps against the validation sets.
+func (v *ValidationSets) CheckInstalledSnaps(snaps []*InstalledSnap) (invalid map[string]bool) {
```

Comment:
this should return nil or a structured error  similar to ValidationSetsConflictError

carrying snaps organized/grouped as either  Invalid, AtInvalidRevision,  Missing with pointers to which validation-sets make them so

happy to chat more